daemons/attrd/attrd_sync.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/msg_xml.h>
+#include <crm/common/attrd_internal.h>
+
+#include "pacemaker-attrd.h"
+
+const char *
+attrd_request_sync_point(xmlNode *xml)
+{
+    if (xml_has_children(xml)) {
+        for (xmlNode *child = first_named_child(xml, XML_ATTR_OP); child != NULL;
+             child = crm_next_same_xml(child)) {
+            const char *sync_point = crm_element_value(child, PCMK__XA_ATTR_SYNC_POINT);
+
+            if (sync_point != NULL) {
+                return sync_point;
```

Comment:
Note that there's an assumption here - each child must have the same sync point, and that if one child has a sync point then all children will be handled as if they have a sync point.  Perhaps that is worth calling out with a comment.

Reply:
Also, note that we put in all sorts of caveats when we added multiple-attribute stuff.  One of the biggest was that it could only work for messages passed from daemon to daemon.  Thus, that is all that's supported here too.  There's no way to say "here's attribute A, here's attribute B, lemme know when they are set" from the command line.

Reply:
Makes sense. We might add command-line support eventually, but I don't think we should allow different sync points.